fzf-tab.zsh

```diff
@@ -306,7 +308,24 @@ _fzf_tab_complete() {
 
 zle -C _fzf_tab_complete complete-word _fzf_tab_complete
 
+_fzf_tab_try_custom_completion() {
+    # do not steal fzf's completions
+    [[ $LBUFFER =~ ${(q)FZF_COMPLETION_TRIGGER-'**'}$ ]] && return 1
+    local tokens=(${(z)LBUFFER})
+    [[ ${LBUFFER[-1]} = ' ' ]] && tokens+=("")
+    local cmd=${tokens[1]}
+    if (( $+functions[${FZF_TAB_CUSTOM_COMPLETIONS_PREFIX}${cmd}] )); then
+        local prefix=${tokens[-1]}
+        local lbuf
+        [ -z "${tokens[-1]}" ] && lbuf=$LBUFFER || lbuf=${LBUFFER:0:-${#tokens[-1]}}
+        prefix="$prefix" eval _fzf_complete_${cmd} ${(q)lbuf}
+        return 0
+    fi
+    return 1
+}
+
 fzf-tab-complete() {
+    [[ -n $FZF_TAB_CUSTOM_COMPLETIONS ]] && _fzf_tab_try_custom_completion && return
```

Comment:
Should it be `(( $FZF_TAB_CUSTOM_COMPLETIONS ))` or `[[ $FZF_TAB_CUSTOM_COMPLETIONS == 1 ]]` here?

Or can we just check if `$FZF_TAB_CUSTOM_COMPLETIONS` exists? ( set to non-null value to enable, set to empty value to disable )

Reply:
should be (( $FZF_TAB_CUSTOM_COMPLETIONS ))

Reply:
It must not have the dollar sign. This is correct: `(( FZF_TAB_CUSTOM_COMPLETIONS ))`. With the dollar sign you are also supporting indirection that you don't want, and potentially even executing arbitrary code.

Reply:
changed